ex04.ex

```diff
@@ -55,7 +55,10 @@ defmodule Ex04 do
 
   """
 
-  def min . . . "your code"
+  def _min(true, a, _b),  do: a
+  def _min(false, _a, b), do: b
+
+  def min([ h | t ]), do: reduce(t, h, &((&1 < &2) |> _min(&1, &2)))
```

Comment:
This works and there's no deduction. But I have a philosophical point to make.

Your code on line 61 does two things: it performs the reduce, and it checks if a is greater then b. Then the functions on 58/9 take a boolean and two numbers and select one of those numbers based on the boolean.

So the "minimum of two numbers" code is split between the comparison on 61 and the derivation of the result on 58/9.

I think it would be a better design to have all the minimum code in one place, so

~~~ elixir
def min_of(a, b) do  # returns a or b
  # ...
end

def min([ h | t ]), do: reduce(t, h, &min_of/2)
end
~~~

Reply:
Oh ok, that makes sense. I was trying to avoid using if else statements so that was one way that I figured I could do in order to avoid that. Will be more vigilant for things like this in the future.